FuzzyAutocomplete/DVTTextCompletionSession+FuzzyAutocomplete.m

```diff
@@ -22,6 +22,8 @@
 #import "FAItemScoringMethod.h"
 #import <objc/runtime.h>
 
+#define dispatch_on_main($block) (dispatch_get_current_queue() == dispatch_get_main_queue() ? $block() : dispatch_sync(dispatch_get_main_queue(), $block))
```

Comment:
http://stackoverflow.com/questions/20860997/dispatch-queue-set-specific-vs-getting-the-current-queue/20862887#20862887 might dissuade you from this